pkg/hive/manager_test.go

```diff
@@ -179,71 +181,207 @@ func TestIsClusterInstallationComplete(t *testing.T) {
+				},
+			},
+			Spec: hivev1.ClusterProvisionSpec{
+				InstallLog: &installLog,
+			},
+		}
+	}
+
 	for _, tt := range []struct {
 		name       string
-		cd         kruntime.Object
+		cd         *hivev1.ClusterDeployment
+		cp         *hivev1.ClusterProvision
 		wantResult bool
-		wantErr    string
+		wantErr    error
 	}{
 		{
 			name: "is installed",
-			cd: &hivev1.ClusterDeployment{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      ClusterDeploymentName,
-					Namespace: fakeNamespace,
-				},
-				Spec: hivev1.ClusterDeploymentSpec{
-					Installed: true,
-				},
-				Status: hivev1.ClusterDeploymentStatus{
-					Conditions: []hivev1.ClusterDeploymentCondition{
-						{
-							Type:   hivev1.ProvisionFailedCondition,
-							Status: corev1.ConditionFalse,
-						},
-					},
+			cd: makeClusterDeployment(
+				true,
+				hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.ProvisionFailedCondition,
+					Status: corev1.ConditionFalse,
 				},
-			},
+			),
 			wantResult: true,
 		},
 		{
 			name: "is not installed yet",
-			cd: &hivev1.ClusterDeployment{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      ClusterDeploymentName,
-					Namespace: fakeNamespace,
+			cd: makeClusterDeployment(
+				false,
+				hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.ProvisionFailedCondition,
+					Status: corev1.ConditionFalse,
+				},
+			),
+			wantResult: false,
+		},
+		{
+			name: "has failed provisioning - no Reason",
+			cd: makeClusterDeployment(
+				false,
+				hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.ProvisionFailedCondition,
+					Status: corev1.ConditionTrue,
 				},
-				Spec: hivev1.ClusterDeploymentSpec{
-					Installed: false,
+			),
+			wantErr:    genericErr,
+			wantResult: false,
+		},
+		{
+			name: "has failed provisioning - Known Reason not relevant to ARO",
+			cd: makeClusterDeployment(
+				false,
+				hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.ProvisionFailedCondition,
+					Status: corev1.ConditionTrue,
+					Reason: "AWSInsufficientCapacity",
 				},
-				Status: hivev1.ClusterDeploymentStatus{
-					Conditions: []hivev1.ClusterDeploymentCondition{
+			),
+			wantErr:    genericErr,
+			wantResult: false,
+		},
+		{
+			name: "has failed provisioning - UnknownError",
+			cd: makeClusterDeployment(
+				false,
+				hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.ProvisionFailedCondition,
+					Status: corev1.ConditionTrue,
+					Reason: "UnknownError",
+				},
+			),
+			wantErr:    genericErr,
+			wantResult: false,
+		},
+		{
+			name: "has failed provisioning - RequestDisallowedByPolicy",
+			cd: makeClusterDeployment(
+				false,
+				hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.ProvisionFailedCondition,
+					Status: corev1.ConditionTrue,
+					Reason: failure.AzureRequestDisallowedByPolicy.Reason,
+				},
+			),
+			cp: makeClusterProvision(`level=info msg=running in local development mode
```

Comment:
Do we need the whole cluster provision log, or can we just specify the error that matches your regex?